### PR TITLE
fix: [CDS-71718]: Artifact Trigger Configuration Tab enhacement

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
#### Screenshots
**Create Artifact Trigger:**
Before:
Screenshot 2023-07-20 at 3 27 31 PM

After:
Screenshot 2023-07-20 at 3 25 25 PM

Edit Artifact Trigger:

Before:
Screenshot 2023-07-20 at 3 27 18 PM
After:
Screenshot 2023-07-20 at 3 26 03 PM

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by pankajsingh-harness** [comment]
retrigger titlecheck

**Review by srdjan-harness — DISMISSED** [review]

File: src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/__tests__/ArtifactTriggerConfigPanel.test.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+          },
+          eventConditions: [],
+          metaDataConditions: []
+        }}
+      />
+    )
+
+    await waitFor(() => {
+      expect(queryByText('name')).toBeInTheDocument()
+    })
```
Comment (rameshx) [inline src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/__tests__/ArtifactTriggerConfigPanel.test.tsx:78]: you can use `await findBy*` here and `waitFor` can be removed

File: src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/__tests__/ArtifactTriggerConfigPanel.test.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+    expect(queryByText('Id')).toBeInTheDocument()
+    expect(queryByPlaceholderText('common.namePlaceholder')).toBeInTheDocument()
+    expect(queryByText('pipeline.artifactTriggerConfigPanel.listenOnNewArtifact')).toBeInTheDocument()
+    expect(queryByText('pipeline.artifactTriggerConfigPanel.artifact')).toBeInTheDocument()
+    expect(queryByText('pipeline.artifactTriggerConfigPanel.defineArtifactSource')).toBeInTheDocument()
```
Comment (rameshx) [inline src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/__tests__/ArtifactTriggerConfigPanel.test.tsx:84]: All of these should use `getBy*` as we are expecting them to be present.

File: src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/__tests__/ArtifactTriggerConfigPanel.test.tsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    // Artifact Source Values
+    expect(queryByText('connectors.GCR.name')).toBeInTheDocument()
+    expect(queryByText(connectorRef)).toBeInTheDocument()
+    expect(queryByText(imagePath)).toBeInTheDocument()
+
+    // Edit artifact source button
+    expect(container.querySelectorAll('[data-icon="Edit"]')).toHaveLength(1)
+
+    // Delete artifact source button
+    expect(container.querySelectorAll('[data-icon="main-trash"]')).toHaveLength(1)
```
Comment (rameshx) [inline src/modules/72-triggers/components/steps/ArtifactTriggerConfigPanel/__tests__/ArtifactTriggerConfigPanel.test.tsx:159]: same as above

**Review by rameshx — APPROVED** [review]

**Pull request merged**